library/network/test/y2firewall/firewalld_test.rb

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+#
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../test_helper"
+require "y2firewall/firewalld"
+
+Yast.import "PackageSystem"
+Yast.import "Service"
+
+describe Y2Firewall::Firewalld do
+  let(:firewalld) { described_class.instance }
+
+  describe "#installed?" do
+    it "returns true it the firewalld is installed" do
```

Comment:
"true *if* ~the~ firewalld"

Reply:
Just a suggestion for the future: I know that using too many contexts can be cumbersome, but what about this?

```ruby
describe "#installed?" do
  before do # this is common code
    allow(Yast::Service).to receive("Enabled").with(described_class::SERVICE).and_return(installed)
  end

  context "when firewalld is installed" do
    let(:installed) { true }

    it "returns true" do
      expect(firewalld).to be_enabled # or expect(firewalld.enabled?).to eq(true)
    end
  end

  context "when firewalld is not installed" do
    let(:installed) { false }

    it "returns false" do
      expect(firewalld).to_not be_enabled # or expect(firewalld.enabled?).to eq(false)
    end
  end
end
```

Reply:
As it was a context with only one test I didn't use it, but it probably make the test more readable.